app/lib/dfe/bigquery/application_metrics.rb

```diff
@@ -0,0 +1,46 @@
+module DfE
+  module Bigquery
+    class ApplicationMetrics
+      attr_accessor :number_of_candidates_submitted_to_date,
+                    :number_of_candidates_submitted_to_same_date_previous_cycle,
+                    :number_of_candidates_with_offers_to_date,
+                    :number_of_candidates_with_offers_to_same_date_previous_cycle,
+                    :number_of_candidates_accepted_to_date,
+                    :number_of_candidates_accepted_to_same_date_previous_cycle,
+                    :number_of_candidates_who_had_all_applications_rejected_this_cycle_to_date,
+                    :number_of_candidates_who_had_all_applications_rejected_this_cycle_to_same_date_previous_cycle,
+                    :number_of_candidates_with_reconfirmed_offers_deferred_from_previous_cycle_to_date,
+                    :number_of_candidates_with_reconfirmed_offers_deferred_from_previous_cycle_to_same_date_previous_cycle,
+                    :number_of_candidates_with_deferred_offers_from_this_cycle_to_date,
+                    :number_of_candidates_with_deferred_offers_from_this_cycle_to_same_date_previous_cycle,
+                    :number_of_candidates_with_all_accepted_offers_withdrawn_this_cycle_to_date,
+                    :number_of_candidates_with_all_accepted_offers_withdrawn_this_cycle_to_same_date_previous_cycle,
+                    :number_of_candidates_who_did_not_meet_any_offer_conditions_this_cycle_to_date,
+                    :number_of_candidates_who_did_not_meet_any_offer_conditions_this_cycle_to_same_date_previous_cycle,
+                    :first_date_in_week,
+                    :last_date_in_week,
+                    :cycle_week
+
+      def initialize(attributes)
```

Comment:
You can achieve something similar if you `include ActiveModel::AttributeAssignment` and do
```ruby
def initialize(attributes)
  assign_attributes(attributes) if attributes
end
```

https://apidock.com/rails/v4.0.2/ActiveRecord/AttributeAssignment/_assign_attribute

However, it might not be as obvious for those who are not familiar with that method.